tests/unit/test_logging.py

```diff
@@ -216,3 +249,36 @@ def test_broken_pipe_in_stdout_flush(self):
         # Sanity check that the log record was written, since flush() happens
         # after write().
         assert output.startswith('my error')
+
+    @pytest.mark.skipif(sys.platform != "win32", reason="Windows only")
+    @pytest.mark.skipif(colorama is None, reason="colorama required")
+    @pytest.mark.parametrize("color", ["always", "auto"])
+    def test_emit_with_color_sets_windows_console(
+        self, capsys, empty_log_record, SetConsoleTextAttribute, color
+    ):
+        """
+        It calls SetConsoleTextAttribute on Windows for colored output.
+        """
+        from pip._vendor.colorama import win32
+
+        with capsys.disabled():
+            if not win32.winapi_test():
+                pytest.skip("output is not connected to a terminal")
```

Comment:
Is there a way to not skip a test in the middle of itself? Maybe just return or something?
I may be wrong on this :smile:

Reply:
IMHO it might be a better idea to move the check out as a decorator, that'd be easier to understand.

Reply:
Yeah, there is a `skipif(colorma is None)`
You can add/replace in this check

Reply:
The thing is that the check needs to run with output capturing disabled (capsys.disabled) 😳 I think it’s clearer this way, but let me know if I missing something

Reply:
I see, I guess its okay